networkx/algorithms/coloring/greedy_coloring.py

```diff
@@ -262,6 +276,11 @@ def greedy_color(G, strategy=strategy_largest_first, interchange=False):
     """
     colors = {}  # dictionary to keep track of the colors of the nodes
 
+    if nx.utils.is_string_like(strategy):
```

Comment:
How about just

    if strategy in GREEDY_COLOR_STRATEGIES:
        strategy = GREEDY_COLOR_STRATEGIES[strategy]